private/bufpkg/bufplugin/bufplugin.go

```diff
@@ -60,3 +63,122 @@ func NewPlugin(
 ) (Plugin, error) {
 	return newPlugin(version, options, runtimeConfig, imageDigest)
 }
+
+func PluginToProtoPluginLanguage(plugin Plugin) registryv1alpha1.PluginLanguage {
+	language := registryv1alpha1.PluginLanguage_PLUGIN_LANGUAGE_UNSPECIFIED
+	if plugin.Runtime() != nil {
+		if plugin.Runtime().Go != nil {
+			language = registryv1alpha1.PluginLanguage_PLUGIN_LANGUAGE_GO
+		} else if plugin.Runtime().NPM != nil {
+			language = registryv1alpha1.PluginLanguage_PLUGIN_LANGUAGE_NPM
+		}
+	}
+	return language
+}
+
+func PluginRuntimeToProtoRuntimeConfig(pluginRuntime *bufpluginconfig.RuntimeConfig) *registryv1alpha1.RuntimeConfig {
+	if pluginRuntime == nil {
+		return nil
+	}
+	runtimeConfig := &registryv1alpha1.RuntimeConfig{}
+	if pluginRuntime.Go != nil {
+		goConfig := &registryv1alpha1.GoConfig{}
+		goConfig.MinimumVersion = pluginRuntime.Go.MinVersion
+		goConfig.RuntimeLibraries = make([]*registryv1alpha1.GoConfig_RuntimeLibrary, 0, len(pluginRuntime.Go.Deps))
+		for _, dependency := range pluginRuntime.Go.Deps {
+			goConfig.RuntimeLibraries = append(goConfig.RuntimeLibraries, GoRuntimeDependencyToProtoGoRuntimeLibrary(dependency))
+		}
+		runtimeConfig.RuntimeConfig = &registryv1alpha1.RuntimeConfig_GoConfig{GoConfig: goConfig}
+	} else if pluginRuntime.NPM != nil {
+		npmConfig := &registryv1alpha1.NPMConfig{}
+		npmConfig.RuntimeLibraries = make([]*registryv1alpha1.NPMConfig_RuntimeLibrary, 0, len(pluginRuntime.NPM.Deps))
+		for _, dependency := range pluginRuntime.NPM.Deps {
+			npmConfig.RuntimeLibraries = append(npmConfig.RuntimeLibraries, NPMRuntimeDependencyToProtoNPMRuntimeLibrary(dependency))
+		}
+		runtimeConfig.RuntimeConfig = &registryv1alpha1.RuntimeConfig_NpmConfig{NpmConfig: npmConfig}
+	}
+	return runtimeConfig
+}
+
+func ProtoRuntimeConfigToPluginRuntime(config *registryv1alpha1.RuntimeConfig) *bufpluginconfig.RuntimeConfig {
+	if config == nil {
+		return nil
+	}
+	runtimeConfig := &bufpluginconfig.RuntimeConfig{}
+	if config.GetGoConfig() != nil {
+		goConfig := &bufpluginconfig.GoRuntimeConfig{}
+		goConfig.MinVersion = config.GetGoConfig().MinimumVersion
```

Comment:
Although (mostly) equivalent, we might want to use the `Get` helper on each of these secondary accessors (this applies to the `.RuntimeLibraries` calls below, too.
```suggestion
		goConfig.MinVersion = config.GetGoConfig().GetMinimumVersion()
```